src/sentry/models/avatars/control_base.py

```diff
@@ -0,0 +1,25 @@
+from sentry.db.models import BoundedBigIntegerField
+from sentry.models.avatars.base import AvatarBase
+
+
+class ControlAvatarBase(AvatarBase):
+    control_file_id = BoundedBigIntegerField(unique=True, null=True)
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def file_class(cls):
+        from sentry.models import ControlFile
+
+        return ControlFile
+
+    @classmethod
+    def file_fk(cls) -> str:
+        return "control_file_id"
+
+    @classmethod
+    def save_avatar(cls, relation, type, avatar=None, filename=None, color=None):
+        instance = super().save_avatar(cls, relation, type, avatar, filename, color)
+        ...
+        return instance
```

Comment:
Is this still a TODO? The implementation in `AvatarBase` has direct references to `File` so we'll need to do something different there.

Reply:
AvatarBase's behavior should be updated. I think this is no longer necessary.